components/function-controller/design/scaling_functions.md

```diff
@@ -0,0 +1,43 @@
+# Serverless Functions Scaling Modes
+
+## Summary
+Initially, the Serverless API supported the `spec.ScaleConfig` field only. The work flow was as follow:
+- Function resources were defaulted to min/max replicas = 1
+- If the user sets a different scaling config, the function controller will create an HPA resources with the user defined min/max values. 
+- The HPA resource target ref will be the Function runtime deployment.
```

Comment:
We can target function as it's scalable and update the deployment `replicas` using function `spec`

Reply:
We actually discussed this. I didn't like initially, it makes sense because this why we avoid having a conflict with other user-created HPA resources to scale the function externally. Also mapping the work flow would be confusing: function.Spec.ScaleConfig.min/max -> Internal HPA resource - > function.Spec.Replicas - > runtime deployment. spec.Replicas :/

Reply:
This will hide user misconfiguration of scaling.

Reply:
Can you give an example?

Reply:
If user set `scaleConfig` he is aware that it enables scaling. In this situation the user shouldn't create any external scaler. As it can introduce bugs. 
Additionally our CR starting behaving strangely because we will set `scaleConfig` to the same values as `replicas` and will wipe user configuration.
The scaling topic should be well documented but it's in other task.

The same situation: user shouldn't create two external scalers as it can undefined behavior.

Adding such behaviour we hide user misconfiguration of scaling and changing the configuration. 

I think that the workflow you mentioned simplify code which is responsible for managing deployments.